### PR TITLE
remove dependency on pypi docker image

### PR DESCRIPTION
**Pull request by amarts**
* pypi images were added to save time to build on arm platform
  (specially for grpcio package). In latest versions, we see that
  it is taking just 2 mins, instead of 90mins. Hence its good to
  remove the dependency (and keep minimum variables)
* Also add a release tag to builder image, so we can debug it if
  needed.

Signed-off-by: Amar Tumballi <amar@kadalu.io>

### CODE REVIEW

**Review by aravindavk — APPROVED** [review]

**Comment by amarts** [comment]
looks like `kubernetes 17.17` pypi didn't work... same issue as 12.x version. May be something we need to change? I will see if I can refresh the PR without that change, which should suffice.

File: .github/workflows/on-release-tag.yml
Comment (leelavg) [inline .github/workflows/on-release-tag.yml:136]: Will the `builder_version` ever be different from `version`, if they are always same we can use only `version` here and in subsequent yml/Dockerfiles?

File: .github/workflows/on-demand-pypi-source-push.yml
Comment (leelavg) [inline .github/workflows/on-demand-pypi-source-push.yml:1]: Just confirming, this'll not be used anymore after this PR merge right?

**Review by leelavg — COMMENTED** [review]
Just for my understanding, is there a chance that will we cross any limits in docker hub when we keep pushing builder image as well during every merge?

File: .github/workflows/on-demand-pypi-source-push.yml
Comment (amarts) [reply .github/workflows/on-demand-pypi-source-push.yml:1]: Yes. But want to remove them in another PR.

**Comment by amarts** [comment]
> Just for my understanding, is there a chance that will we cross any limits in docker hub when we keep pushing builder image as well during every merge?

We may. But for now, I am considering to get to that issue to resolve that issue. For now, not bother. We can also move to paid service to resolve issues for sometime. Lets face it when we get there.

File: .github/workflows/on-release-tag.yml
Comment (amarts) [reply .github/workflows/on-release-tag.yml:136]: not yet sure. Lets keep it this way for a month or 2 and see where we can optimize, and how the other platform story evolves.

**Pull request merged**
